src/blocks/backlight.rs

```diff
@@ -63,67 +63,38 @@
 //! # Icons Used
 //! - `backlight` (as a progression)
 
-use std::cmp::max;
-use std::ops::Range;
-use std::path::{Path, PathBuf};
-
-use inotify::{Inotify, WatchMask};
-use tokio::fs::{read_dir, OpenOptions};
-use tokio::io::AsyncWriteExt;
+use calibright::{self, config::Config as CalibrightConfig, CalibrightBuilder};
 
 use super::prelude::*;
-use crate::util::read_file;
-
-make_log_macro!(debug, "backlight");
-
-#[zbus::dbus_proxy(
-    interface = "org.freedesktop.login1.Session",
-    default_service = "org.freedesktop.login1",
-    default_path = "/org/freedesktop/login1/session/auto"
-)]
-trait Session {
-    fn set_brightness(&self, subsystem: &str, name: &str, brightness: u32) -> zbus::Result<()>;
-}
-
-/// Location of backlight devices
-const DEVICES_PATH: &str = "/sys/class/backlight";
-
-/// Filename for device's max brightness
-const FILE_MAX_BRIGHTNESS: &str = "max_brightness";
-
-/// Filename for current brightness.
-const FILE_BRIGHTNESS: &str = "actual_brightness";
-
-/// amdgpu drivers set the actual_brightness in a different scale than
-/// [0, max_brightness], so we have to use the 'brightness' file instead.
-/// This may be fixed in the new 5.7 kernel?
-const FILE_BRIGHTNESS_AMD: &str = "brightness";
-
-/// set the requested brightness level
-const FILE_BRIGHTNESS_WRITE: &str = "brightness";
-
-/// Range of valid values for `root_scaling`
-const ROOT_SCALDING_RANGE: Range<f64> = 0.1..10.;
 
 #[derive(Deserialize, Debug, SmartDefault)]
 #[serde(default)]
 pub struct Config {
     device: Option<String>,
     format: FormatConfig,
-    #[default(5)]
-    step_width: u8,
-    #[default(5)]
-    minimum: u8,
-    #[default(100)]
-    maximum: u8,
-    cycle: Option<Vec<u8>>,
-    #[default(1.0)]
-    root_scaling: f64,
+    #[default(5.0)]
+    step_width: f64,
+    #[default(5.0)]
+    minimum: f64,
+    #[default(100.0)]
+    maximum: f64,
+    cycle: Option<Vec<f64>>,
     invert_icons: bool,
-    #[default(1.0)]
-    ddcci_sleep_multiplier: f64,
-    #[default(10)]
-    ddcci_max_tries_write_read: u8,
+    //Calibright config settings
+    root_scaling: Option<f64>,
+    ddcci_sleep_multiplier: Option<f64>,
+    ddcci_max_tries_write_read: Option<u8>,
+}
+
+impl Error {
```

Comment:
@MaxVerevkin , any suggestions on mapping errors from one crate to another?
I am basically using the same error structure as i3status-rs, except without the block.

Reply:
Perhaps using a structured error enum (with `thiserror`) would be better for a library? This would allow crate users to distinguish between error variants.

Reply:
But in general, to map errors just implement `From<calibright::Error> for Error`.

Reply:
I'd tried that but got ```multiple `impl`s satisfying `errors::Error: From<_>` found```

> > I still haven't committed/released anything for calibrigt... Just need to add the bin and config parser.
> 
> I can contribute that if you wish.

I made a simple cli and got config parsing sorted out!